packages/blade/src/components/Menu/_decisions/decisions.md

```diff
@@ -0,0 +1,372 @@
+# Menu
+
+Action Menu displays a list of actions on temporary surfaces. They allow users to action(s) from multiple options. They appear when users interact with a button, action, or other control.
+
+Unlike Dropdown, they are specifically made to be non-selectable, clickable items and are more flexible with what they can contain. Checkout [Menu vs Dropdown](#menu-vs-dropdown) for more context.
+
+<img alt="Menu Overview Image" src="overview.png" width="300px" />
+
+## Design
+
+- [Figma - Menu](https://www.figma.com/design/jubmQL9Z8V7881ayUD95ps/Blade-DSL?node-id=90082-41948&m=dev)
+
+## API
+
+```jsx
+<Menu>
+  {/* Any interactive item which has onClick, onMouseDown exposed */}
+  <Button>Open Menu</Button>
+  <MenuOverlay>
+    <MenuHeader title="" />
+    {/* All of this is a slot */}
+    <Box display="flex" paddingY="spacing.4" gap="spacing.2">
+      <Avatar name="Saurabh Daware" />
+      <Text>Saurabh Daware</Text>
+    </Box>
+    <MenuDivider />
+    <MenuItem title="Undo" onClick={() => console.log('Undo')} />
+    <MenuItem title="Redo" isDisabled />
+    <MenuItem title="Cut" />
+    {/* Sub Menu */}
+    <Menu>
+      <MenuItem title="Copy" />
```

Comment:
What if we use a button/trigger here? Would it throw an error?

Reply:
Nope. It will use that button is submenu trigger. Menus are flexible so there might be usecase of some custom component triggering submenu